### PR TITLE
Apply tiling on parallel dims and reduction dims separately.

### PR DESCRIPTION
**Pull request by hanhanW**
We tried to handle padding issue and came out with a workaround which
might generates redundant one-trip loops. After prototype more, we found
that there are non-trivial amount of work, see
https://github.com/google/iree/issues/8228.

The PR disable padding options and make it have same behaviors as
sandbox.

### CODE REVIEW

**Comment by MaheshRavishankar** [comment]
Could we also update the verifier to check that the tiling sizes for the fused stage doesnt have the reduction dims tile sizes set, and the converse is true for the stage that tiles the reduction dimensions.

File: iree/compiler/Codegen/LLVMCPU/Passes.cpp [hunk 1]
```diff
@@ -157,13 +157,29 @@ LogicalResult verifyDoubleTilingExpertPassPipelineConfig(
     }
   }
 
+  int numLoops = interfaceOp.getNumLoops();
+  SmallVector<int64_t> secondLevelTileSizes = loweringConfig.getTileSizeVals(
+      static_cast<unsigned>(TilingLevel::L1Tiles));
+  if (secondLevelTileSizes.size() != numLoops) {
```
Comment (MaheshRavishankar) [inline iree/compiler/Codegen/LLVMCPU/Passes.cpp:163]: Based on overall comment on the PR. We need to check that the tile sizes are specified only for the parallel loops.

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]

File: iree/compiler/Codegen/LLVMCPU/Passes.cpp (hunk 1 above)
Comment (hanhanW) [reply iree/compiler/Codegen/LLVMCPU/Passes.cpp:163]: Do we get parallel loops by `PartitionableLoopsInterface::getPartitionableLoops` method?
The method takes `maxNumPartitionedLoops` argument. In this case, do we pass INF? (or maybe we can add another method to the interface)

Comment (MaheshRavishankar) [reply iree/compiler/Codegen/LLVMCPU/Passes.cpp:163]: You should be able to use this https://github.com/google/iree/blob/15fce0a6c29b7eb810e6ad36706293dcb855148d/iree/compiler/Codegen/Utils/Utils.h#L21

(and yes you should be using the `getPartitionableLoops` method (at least for now).

Comment (hanhanW) [reply iree/compiler/Codegen/LLVMCPU/Passes.cpp]: getPartitionableLoops is a bit different from what I thought. It also drops unit dims and makes logic a bit complicated, which is fine I think.

**Review by MaheshRavishankar — APPROVED** [review]

**Pull request merged**